File: data/generated_projects/Calculadora/com/example/Calculadora.java
package com.example;

import java.util.Scanner;

public class Calculadora {
 
    private static final String MENU_SUMA = "1. Suma";
    private static final String MENU_RESTA = "2. Resta";
    private static final String MENU_MULTIPLICACION = "3. Multiplicación";
    private static final String MENU_DIVISION = "4. División";
    private static final String MENU_SALIR = "5. Salir";

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean continuar = true;
        
        while (continuar) {
            mostrarMenu();
            int opcion = leerOpcion(scanner);
            
            if (opcion == 5) {
                continuar = false;
                System.out.println("Gracias por usar la calculadora.");
                continue;
            }
            
            try {
                double numero1 = leerNumero(scanner, "Ingrese el primer número:");
                double numero2 = leerNumero(scanner, "Ingrese el segundo número:");
                double resultado = realizarOperacion(opcion, numero1, numero2);
                System.out.println("Resultado: " + resultado);
            } catch (ArithmeticException e) {
                System.out.println("Error: No se puede dividir por cero.");
            } catch (NumberFormatException e) {
                System.out.println("Error: Ingrese un número válido.");
            }
            
            System.out.print("¿Desea realizar otra operación? (s/n): ");
            continuar = scanner.nextLine().toLowerCase().startsWith("s");
        }
        
        scanner.close();
    }

    private static void mostrarMenu() {
        System.out.println("\n=== Menú de Operaciones ===");
        System.out.println(MENU_SUMA);
        System.out.println(MENU_RESTA);
        System.out.println(MENU_MULTIPLICACION);
        System.out.println(MENU_DIVISION);
        System.out.println(MENU_SALIR);
        System.out.print("Seleccione una opción: ");
    }

    private static int leerOpcion(Scanner scanner) {
        while (true) {
            String input = scanner.nextLine();
            if (input.matches("[1-5]")) {
                return Integer.parseInt(input);
            }
            System.out.print("Opción inválida. Intente nuevamente: ");
        }
    }

    private static double leerNumero(Scanner scanner, String mensaje) {
        while (true) {
            System.out.print(mensaje + " ");
            String input = scanner.nextLine();
            if (input.matches("-?[0-9]+([.][0-9]+)?")) {
                return Double.parseDouble(input);
            }
            System.out.println("Entrada inválida. Por favor ingrese un número.");
        }
    }

    private static double realizarOperacion(int opcion, double a, double b) {
        switch (opcion) {
            case 1: return a + b;
            case 2: return a - b;
            case 3: return a * b;
            case 4:
            if (b == 0) {
                System.out.println("Error: No se puede dividir por cero.");
                return 0.0;
            }
            return a / b;
            default: throw new IllegalArgumentException("Opción inválida");
        }
    }
}

File: data/generated_projects/Calculadora/Calculadora.cs
using System;

namespace Calculadora
{
    class Program
    {
        static void Main(string[] args)
        {
            bool continuar = true;
            while (continuar)
            {
                MostrarMenu();
                int opcion = LeerOpcion();

                if (opcion == 5)
                {
                    continuar = false;
                    Console.WriteLine("Gracias por usar la calculadora.");
                    continue;
                }

                Console.Write("Ingrese primer número: ");
                double num1 = ConvertirADouble();
                Console.Write("Ingrese segundo número: ");
                double num2 = ConvertirADouble();

                double resultado = 0;
                try
                {
                    switch (opcion)
                    {
                        case 1:
                            resultado = Sumar(num1, num2);
                            break;
                        case 2:
                            resultado = Restar(num1, num2);
                            break;
                        case 3:
                            resultado = Multiplicar(num1, num2);
                            break;
                        case 4:
                            resultado = Dividir(num1, num2);
                            break;
                        case 5:
                            // Ya manejado anteriormente
                            break;
                        default:
                            Console.WriteLine("Opción inválida.");
                            continue;
                    }
                }
                catch (DivideByZeroException)
                {
                    Console.WriteLine("Error: No se puede dividir por cero.");
                    continue;
                }

                Console.WriteLine($"Resultado: {resultado}");
                Console.WriteLine("Presione cualquier tecla para continuar...");
                Console.ReadKey();
            }
        }

        static void MostrarMenu()
        {
            Console.Clear();
            Console.WriteLine("=== Calculadora ===");
            Console.WriteLine("1. Sumar");
            Console.WriteLine("2. Restar");
            Console.WriteLine("3. Multiplicar");
            Console.WriteLine("4. Dividir");
            Console.WriteLine("5. Salir");
            Console.Write("Seleccione una opción: ");
        }

        static int LeerOpcion()
        {
            while (true)
            {
                if (int.TryParse(Console.ReadLine(), out int opcion) && opcion >= 1 && opcion <= 5)
                    return opcion;
                Console.WriteLine("Opción inválida. Intente nuevamente.");
            }
        }

        static double ConvertirADouble()
        {
            while (true)
            {
                if (double.TryParse(Console.ReadLine(), out double valor))
                    return valor;
                Console.WriteLine("Número inválido. Intente nuevamente.");
            }
        }

        static double Sumar(double a, double b) => a + b;
        static double Restar(double a, double b) => a - b;
        static double Multiplicar(double a, double b) => a * b;
        static double Dividir(double a, double b)
        {
            if (b == 0) throw new DivideByZeroException();
            return a / b;
        }
    }
}